ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTableConfirmation/NetworkTableConfirmation.tsx

```diff
@@ -84,45 +85,76 @@ const NetworkTableConfirmation = ({
       />
     );
   } else if (
-    expanded?.content === ExpandedState.MARK_CONNECTED ||
-    expanded?.content === ExpandedState.MARK_DISCONNECTED
+    [
+      ExpandedState.MARK_CONNECTED,
+      ExpandedState.MARK_DISCONNECTED,
+      ExpandedState.DISCONNECTED_WARNING,
+    ].includes(expanded?.content)
   ) {
-    const markConnected = expanded?.content === ExpandedState.MARK_CONNECTED;
+    const showDisconnectedWarning =
+      expanded?.content === ExpandedState.DISCONNECTED_WARNING;
+    const markConnected =
+      expanded?.content === ExpandedState.MARK_CONNECTED ||
+      showDisconnectedWarning;
     const event = markConnected ? "connected" : "disconnected";
+    const updateConnection = () => {
+      dispatch(
+        machineActions.updateInterface({
+          interfaceId: nic?.id,
+          linkConnected: !!markConnected,
+          systemId: machine.system_id,
+        })
+      );
+    };
+    let message: ReactNode;
+    if (showDisconnectedWarning) {
+      message = (
+        <>
+          This interface is <strong>disconnected</strong>, it cannot be
+          configured unless a cable is connected.
+          <br />
+          If this is no longer true,
```

Comment:
Just need to add a space here.

Also I feel like this is a very weird pattern. The link text makes it look like it should take you to another page. Do you know if this type of thing is done for any of the other network actions? If not I reckon we should just leave it as text here and keep the submit button as the only way to perform the action.

Reply:
Yeah, good idea, that'd be an improvement I think.